backend/src/test/kotlin/cz/loono/backend/api/controller/ExaminationsControllerTest.kt

```diff
@@ -0,0 +1,98 @@
+import org.springframework.boot.jdbc.EmbeddedDatabaseConnection
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.boot.test.context.TestConfiguration
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Import
+import org.springframework.context.annotation.Primary
+import java.time.Clock
+import java.time.Instant
+
+@DataJpaTest
+@Import(value = [ExaminationRecordService::class, PreventionService::class])
+@AutoConfigureTestDatabase(connection = EmbeddedDatabaseConnection.H2)
+class ExaminationsControllerTest(
+    private val recordService: ExaminationRecordService,
+    private val preventionService: PreventionService,
+    private val repo: AccountRepository
+) {
+
+    @Test
+    fun `Should add badge and points`() {
+        val controller = ExaminationsController(recordService, preventionService)
+        val basicUser = createBasicUser()
+        var existingAccount = createAccount()
+        val examinationRecord = ExaminationRecordDto(type = ExaminationTypeEnumDto.DENTIST)
+        // This is done to get assigned ID by the DB
+        existingAccount = repo.save(existingAccount)
+        val expectedBadge = Badge(BadgeTypeDto.HEADBAND.value, existingAccount.id, 1, Instant.EPOCH.toOffsetDateTime(), existingAccount)
+
+        var examUUID = controller.updateOrCreate(basicUser, examinationRecord).uuid!!
+        controller.confirm(basicUser, ExaminationTypeEnumDto.DENTIST.toString(), ExaminationIdDto(examUUID))
+
+        var actual = repo.findByUid("uid")!!
+        assertThat(actual.badges).containsExactly(expectedBadge)
+        assertThat(actual.points).isEqualTo(300)
+
+        controller.confirm(basicUser, ExaminationTypeEnumDto.DENTIST.toString(), ExaminationIdDto(examUUID))
+
+        // Making sure that level upgraded and points increased
+        actual = repo.findByUid("uid")!!
+        assertThat(actual.badges).containsExactly(expectedBadge.copy(level = 2))
+        assertThat(actual.points).isEqualTo(600)
+
+        // Creating exam of another type
+        examUUID = controller.updateOrCreate(basicUser, examinationRecord.copy(type = ExaminationTypeEnumDto.UROLOGIST)).uuid!!
+        controller.confirm(basicUser, ExaminationTypeEnumDto.UROLOGIST.toString(), ExaminationIdDto(examUUID))
+
+        assertThat(actual.badges).containsExactly(
+            expectedBadge.copy(level = 2), expectedBadge.copy(type = BadgeTypeDto.BELT.value)
+        )
+        assertThat(actual.points).isEqualTo(900)
+    }
+
+    @Test
+    fun `Should fail when unsupported exam confirmed`() {
+        val controller = ExaminationsController(recordService, preventionService)
+        val basicUser = createBasicUser()
+        val existingAccount = createAccount()
+        val examinationRecord = ExaminationRecordDto(type = ExaminationTypeEnumDto.MAMMOGRAM)
```

Comment:
@rposkocil btw, I think we should introduce validation on BE side, because now a male user can register for `MAMMOGRAM` examination or is it expected?

Reply:
Good idea. Let's do it.

Reply:
it's in the issue here https://github.com/cesko-digital/loono-be/issues/32